lib.js

```diff
@@ -0,0 +1,28 @@
+function computeTitle(title) {
+  const getting = browser.storage.sync.get({
+    prefix: '',
+    prefixGlue: '',
+  });
+
+  return getting.then((result) => {
+    const computedTitle = result.prefix
+      + ((result.prefix && title && result.prefixGlue) || '')
+      + (title || '');
+
+    return computedTitle ? `[${computedTitle}] ` : '';
+  });
+}
+
+function setWindowTitle(windowId) {
+  return browser.sessions.getWindowValue(windowId, 'title')
+    .catch(() => '')
```

Comment:
So this catches errors if windowId is not present in the session storage 
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/sessions/getWindowValue#Return_value

If that's the purpose it should fail instead of proceeding with setting the computed title for a nonexistant windowId.

Reply:
actually the purpose was to solve a problem with undefined title, but obviously I didn't read the documentation. It should be:
`.then(windowTitle => windowTitle || '')`